pygmt/clib/session.py

```diff
@@ -1474,6 +1475,7 @@ def virtualfile_from_data(
         z=None,
         extra_arrays=None,
         required_z=False,
+        required_data=True,
```

Comment:
Following the [Code Style](https://www.pygmt.org/dev/contributing.html#code-style), no underscore should be used for words bridged only by consonants. However, already `required_z` and `check_kind` contain underscores...
```suggestion
        required_z=False,
        requireddata=True,
```

Reply:
These are internal functions for developers/maintainers. So maybe we don't have to strictly follow the code style?

Reply:
That code style is meant for GMT aliases, so we can use `required_data` here.

Reply:
OK.